characteristic_secure/web_app.js

```diff
@@ -0,0 +1,179 @@
+/**
+ *  Copyright 2021 Google LLC
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+// Espruino devices publish a UART service by default.
+const nordicUARTService = getEspruinoPrimaryService();
+
+const ageServiceNumber = 0x2A80;
+const ageServiceID = '00002a80-0000-1000-8000-00805f9b34fb';
+const ageCharacteristicName = 'age';
+const ageCharacteristicID = '00002a80-0000-1000-8000-00805f9b34fb';
+
+const heartRateServiceName = 'heart_rate';
+const heartRateServiceID = '0000180d-0000-1000-8000-00805f9b34fb';
+const heartRateMaxCharacteristicName = 'heart_rate_max';
+const heartRateMaxCharacteristicID = '00002a8d-0000-1000-8000-00805f9b34fb';
+
+let gattServer = undefined;
+
+/**
+ * Load the device code to the Espruino IDE.
+ */
+function loadEspruinoDeviceCode() {
+  fetch('device_code.js').then(response => response.text()).then(data => {
+    let url = 'http://www.espruino.com/webide?code=' + encodeURIComponent(data);
+    window.open(url, '_window');
+  });
+}
+
+function onGattDisconnected(evt) {
+  const device = evt.target;
+  logInfo(`Disconnected from GATT on device ${device.name}.`);
+  assertFalse(gattServer.connected, 'Server connected');
+}
+
+async function startTest() {
+  clearStatus();
+  logInfo('Starting test');
+
+  $('btn_start_test').disabled = true;
+  $('btn_load_code').disabled = true;
+
+  try {
+
```

Comment:
Extra newline.

Reply:
Missed this. I'll get it before landing.